guide/popular-topics/permissions.md

```diff
@@ -0,0 +1,335 @@
+The library does not handle implicit permissions for you, so understanding how the system works is vital for you as a bot developer.
+
+Let's say you want to send a message in a channel. To prevent unnecessary API calls you want to make sure your bots permissions in this channel include `SEND_MESSAGES` (more on how to achieve this down below). The check passes, but you still can't actually send the message and are greeted with `DiscordAPIError: Missing Access`.
+
+This means your bot is missing `VIEW_CHANNEL` and as such can't send messages either.
+
+One possible scenario causing this: The channel has permission overwrites for the default role @everyone to grant `SEND_MESSAGES` so everyone who can see the channel can also write in it, but at the same time has an overwrite to deny `VIEW_CHANNEL` to make it only accessible to a subset of members.
+
+As you only check for `SEND_MESSAGES` the bot will try to execute the send, but since `VIEW_CHANNEL` is missing, the request is denied by the API.
+
+### Limitations and oddities
+
+- Your bot needs `MANAGE_ROLES` in its base permissions to change base permissions.
+- It needs `MANAGE_ROLES` in its final permissions to change permission overwrites.
+- It can not edit permissions for roles that are higher than or equal to its highest role.
+- It can not grant permissions it doesn't have.
+- It can manage overwrites for roles or users with higher roles than its own highest role.
+- It can manage overwrites for permissions it doesn't have.
+- Members with the `ADMINISTRATOR` permission are not affected by overwrites at all.
+
+### Missing permissions
+
+During your tests you will likely run into `DiscordAPIError: Missing Permissions` at some point. This error can be caused by one of the following:
+
+- Your bot is missing the needed permission to execute this action in it's calculated base or final permissions (requirement changes based on the type of action you are trying to execute).
+- It is trying to execute an action on a guild member with a role higher than or equal to your bots highest role.
+- It is trying to modify a role that is higher than or equal to its highest role.
+- It is trying to execute a forbidden action on the server owner.
+- It is trying to execute an action based on another unfulfilled factor (for example reserved for partnered guilds). 
+
+<tip>The `ADMINISTRATOR` permission being granted does not skip any hierarchical check!</tip>
+
+## Base permissions
+
+### Setting role permissions
+
+Permissions are set on roles, not the guild member itself. To change them you access the Role object (for example via `guild.roles`) and use the `.setPermissions()` method.
+Let's change the base permissions for @everyone so you can see it in action:
+
+```js
+// permission flag array
+guild.defaultRole.setPermissions(['SEND_MESSAGES', 'VIEW_CHANNEL']);
+
+// permission bit field in decimal representation
+guild.defaultRole.setPermissions(3072);
```

Comment:
Unsure if including the bit field part is necessary, but I suppose some may find it useful. I think that, at the very least, a link to one of the [permissions calculator sites](https://discordapi.com/permissions.html) would be ideal. Probably in a `<tip>` somewhere above or below this example.

Reply:
removing bitfields from examples and adding a <tip> in terminology instead with "you can use permission decimals where we use flag literals" with the link to the dev portal.

The official permission calculator moved to the "Bot" page in the dev portal and I think we should stick with the official one.